fuel-core/src/executor.rs

```diff
@@ -266,6 +268,18 @@ impl Executor {
         Ok(())
     }
 
+    /// Verify the transaction has at least one coin.
+    ///
+    /// TODO: This verification really belongs in fuel-tx, and can be removed once
+    ///       https://github.com/FuelLabs/fuel-tx/issues/118 is resolved.
+    fn verify_tx_has_at_least_one_coin(&self, tx: &Transaction) -> Result<(), Error> {
+        if tx.inputs().iter().filter(|input| input.is_coin()).count() == 0 {
```

Comment:
Optional nit: this looks like `!iter().filter(Input::is_coin).next().map(|_| ()).ok_or(TransactionValidityError::NoCoinInput(tx.id()).into())`

That way, instead of counting the iterators, we just lazy-check the next one - that already satisfies the count != 0 criteria

Reply:
That's maybe slightly less readable but seems like a good technique. Maybe we can do it that way when we impl on the fuel-tx side.